File: impl/js/difftest.js
import {readFileSync} from "fs";
import {normalize} from "./coia.js";
const enc = s => [...s].map(c => c.codePointAt(0).toString(16).toUpperCase()).join(" ");
const dec = l => l.trim() === "" ? "" : l.trim().split(/ +/).map(x => String.fromCodePoint(parseInt(x,16))).join("");
const lines = readFileSync("difftest-input.txt","utf8").split("\n");
for (const l of lines) { if (l === "") continue; console.log(enc(normalize(dec(l)))); }

File: impl/java/DiffTest.java
import java.nio.file.*;
import java.util.*;
import java.util.stream.Collectors;

public class DiffTest {
    public static void main(String[] a) throws Exception {
        for (String line : Files.readAllLines(Paths.get("difftest-input.txt"))) {
            if (line.isEmpty()) continue;
            StringBuilder in = new StringBuilder();
            for (String h : line.trim().split(" +"))
                in.appendCodePoint(Integer.parseInt(h, 16));
            String r = Coia.normalize(in.toString());
            System.out.println(r.codePoints()
                .mapToObj(cp -> Integer.toHexString(cp).toUpperCase())
                .collect(Collectors.joining(" ")));
        }
    }
}
